generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/Combined.java

```diff
@@ -0,0 +1,84 @@
+package com.scottlogic.deg.generator.walker.reductive;
+
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+public class Combined<T> {
+
+    private final T value;
+
+
+    /**
+     * Returns an {@code Combined} with the specified present non-null value.
+     *
+     * @param <T> the class of the value
+     * @param value the value to be present, which must be non-null
+     * @return an {@code Combined} with the value present
+     * @throws NullPointerException if value is null
+     */
+    public static <T> Combined<T> of(T value) {
+        return new Combined<>(value);
+    }
+
+
+    /**
+     * Return {@code true} if there is no value present, otherwise {@code false}.
+     *
+     * @return {@code true} if there is no value present, otherwise {@code false}
+     */
+    public boolean isContradictory() {
+        return value == null;
+    }
+
+    /**
+     * If a value not Contradictory in this {@code Combined}, returns the value,
+     * otherwise throws {@code NoSuchElementException}.
+     *
+     * @return the non-null value held by this {@code Optional}
+     * @throws NoSuchElementException if is Contradictory
+     *
+     * @see Combined#isContradictory() ()
+     */
+    public T get() {
+        if (value == null) {
+            throw new NoSuchElementException("No value present");
+        }
+        return value;
+    }
+
+    /**
+     * Returns a Contradictory {@code Combined} instance.  No value is present for this
+     * Combined.
+     *
+     * @param <T> Type of the non-existent value
+     * @return a Contradictory {@code Combined}
+     */
+    public static<T> Combined<T> contradictory() {
+        @SuppressWarnings("unchecked")
```

Comment:
Why not
```
return new Combined();
```

Then you can remove the static field and the warnings and weird casts here